sdcm/cluster_k8s/__init__.py

```diff
@@ -839,6 +864,26 @@ def deploy_scylla_cluster(self, node_pool: CloudK8sNodePool = None, node_prepare
         self.init_scylla_config_map()
         self.scylla_restart_required = False
 
+        # Deploy NodeConfig resource that will prepare K8S nodes for Scylla pods
+        if self.params.get('k8s_enable_performance_tuning') and self.CREATE_NODE_CONFIG_CR and node_pool:
+            self.kubectl_wait(
+                "--for condition=established crd/scyllaoperatorconfigs.scylla.scylladb.com")
+            self.kubectl_wait("--for condition=established crd/nodeconfigs.scylla.scylladb.com")
+
+            if scylla_utils_docker_image := self.params.get('k8s_scylla_utils_docker_image'):
+                self.kubectl(
+                    "patch scyllaoperatorconfigs.scylla.scylladb.com cluster --type merge "
+                    "-p '{\"spec\":{\"scyllaUtilsImage\":\"%s\"}}'" % scylla_utils_docker_image,
+                    ignore_status=False)
+
+            self.apply_file(
+                "sdcm/k8s_configs/node-config-crd.yaml",
+                modifiers=affinity_modifiers, envsubst=False)
+            # NOTE: no need to wait explicitly for the new objects readiness.
+            #       Scylla pods will get in sync automatically.
+            #       Just sleep for some time to avoid races.
+            time.sleep(5)
```

Comment:
Let's put it into a method

Reply:
Ok, will do.

Reply:
After close looking and attempt to do it I came to the idea that new variant looks more complex because it is tied to the couple of variables which will be needed to be provided. So, left it as is.